packages/react-components/src/components/chart/hooks/useEvents.ts

```diff
@@ -0,0 +1,135 @@
+import { useEffect } from 'react';
+import { EChartsType, getInstanceByDom } from 'echarts';
+import { MAX_TREND_CURSORS, TREND_CURSOR_CLOSE_GRAPHIC_INDEX } from '../eChartsConstants';
+import { calculateTimeStamp } from '../utils/getInfo';
+import { v4 as uuid } from 'uuid';
+import { getNewTrendCursor, ondragUpdateTrendCursor } from '../utils/getTrendCursor';
+import { UseEventsProps } from '../types';
+import useDataStore from '../../../store';
+
+let trendCursorStaticIndex = 0;
+
+const useEvents = ({
+  ref,
+  graphic,
+  size,
+  isInCursorAddMode,
+  setGraphic,
+  viewport,
+  series,
+  yMin,
+  yMax,
+  isInSyncMode,
+  groupId,
+}: UseEventsProps) => {
+  // sync mode actions
+  const addTrendCursorsToSyncState = useDataStore((state) => state.addTrendCursors);
+  const updateTrendCursorsInSyncState = useDataStore((state) => state.updateTrendCursors);
+  const deleteTrendCursorsInSyncState = useDataStore((state) => state.deleteTrendCursors);
+  const mouseoverHandler = (isInCursorAddMode: boolean, chart?: EChartsType) => {
```

Comment:
Why does this function need to be defined here? It looks like it takes all required data through params.

Reply:
I am subscribing and unsubscribing the mouseoverHandler, and the same callback has be to provided in both cases, if i unsubscribe w/o providing this specific callback, echarts will remove all mouseover behaviour which is not what we want.